### PR TITLE
Consolidate logging in SessionOrchestrator

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Consolidates logging into the `SessionOrchestrator & makes the `session` property private in the `BackgroundActivityService`.

## Testing

Relied on existing test coverage.

### CODE REVIEW

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/EmbraceBackgroundActivityService.kt [hunk 1]
```diff
@@ -50,10 +50,9 @@ internal class EmbraceBackgroundActivityService(
     override fun endBackgroundActivityWithState(timestamp: Long) {
         // kept for backwards compat. the backend expects the start time to be 1 ms greater
         // than the adjacent session, and manually adjusts.
-        val activity = backgroundActivity ?: return
         val message = stopCapture(
             FinalEnvelopeParams.BackgroundActivityParams(
-                initial = activity,
+                initial = backgroundActivity ?: return,
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/EmbraceBackgroundActivityService.kt:55]: What does putting a return here do?

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/orchestrator/SessionOrchestratorImpl.kt [hunk 2]
```diff
@@ -110,6 +125,20 @@ internal class SessionOrchestratorImpl(
...
             backgroundActivityService?.endBackgroundActivityWithCrash(timestamp, crashId)
+            logger.logDebug("Session ended with crash")
         }
     }
+
+    private fun logSessionStateChange(
+        sessionId: String?,
+        timestamp: Long,
+        inBackground: Boolean,
+        stateChange: String
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/orchestrator/SessionOrchestratorImpl.kt:136]: Is this just for logging or do you intent this to be used elsewhere?

Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/orchestrator/SessionOrchestratorImpl.kt:136]: I assume this is just something you need for logging and not a concept you want to leverage more later?

**Review by bidetofevil — APPROVED** [review]
LGTM. Just a couple of questions for my own clarification

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/EmbraceBackgroundActivityService.kt (hunk 1 above)
Comment (fractalwrench) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/EmbraceBackgroundActivityService.kt:55]: It does return from the method. However, I'll switch it back as it was clearer before

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/orchestrator/SessionOrchestratorImpl.kt (hunk 2 above)
Comment (fractalwrench) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/orchestrator/SessionOrchestratorImpl.kt:136]: Just logging for now. Ideally I'd like to keep a record of all session state changes in some sort of buffer for debugging purposes, but that will come later.

**Pull request merged**